src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js

```diff
@@ -63,224 +64,228 @@ const defaultProps = {
+    const [formError, setFormError] = useState({});
+    const [validateCode, setValidateCode] = useState(props.credentials.validateCode || '');
+    const [twoFactorAuthCode, setTwoFactorAuthCode] = useState('');
+    const [linkSent, setLinkSent] = useState(false);
 
-        this.state = {
-            formError: {},
-            validateCode: props.credentials.validateCode || '',
-            twoFactorAuthCode: '',
-            linkSent: false,
-        };
-    }
+    const prevIsVisible = usePrevious(props.isVisible);
+    const prevRequiresTwoFactorAuth = usePrevious(props.account.requiresTwoFactorAuth);
+    const prevValidateCode = usePrevious(props.credentials.validateCode);
 
-    componentDidMount() {
-        if (!canFocusInputOnScreenFocus() || !this.inputValidateCode || !this.props.isVisible) {
+    const inputValidateCodeRef = useRef();
+    const input2FARef = useRef();
+
+    useEffect(() => {
+        if (!inputValidateCodeRef.current || prevIsVisible || !props.isVisible || !canFocusInputOnScreenFocus()) {
             return;
         }
-        this.inputValidateCode.focus();
-    }
+        inputValidateCodeRef.current.focus();
+    }, [props.isVisible, prevIsVisible]);
 
-    componentDidUpdate(prevProps) {
-        if (!prevProps.isVisible && this.props.isVisible) {
-            this.inputValidateCode.focus();
+    useEffect(() => {
+        if (!inputValidateCodeRef.current) {
+            return;
         }
-        if (prevProps.isVisible && !this.props.isVisible && this.state.validateCode) {
-            this.clearValidateCode();
+
+        if (!validateCode) {
+            return;
         }
 
-        // Clear the code input if a new magic code was requested
-        if (this.props.isVisible && this.state.linkSent && this.props.account.message && this.state.validateCode) {
-            this.clearValidateCode();
+        // Clear the code input if magic code valid or a new magic code was requested
+        if ((prevIsVisible && !props.isVisible) || (props.isVisible && linkSent && props.account.message)) {
+            setValidateCode('');
         }
-        if (!prevProps.credentials.validateCode && this.props.credentials.validateCode) {
-            this.setState({validateCode: this.props.credentials.validateCode});
+    }, [props.isVisible, props.account.message, prevIsVisible, linkSent, validateCode]);
+
+    useEffect(() => {
+        if (prevValidateCode || !props.credentials.validateCode) {
+            return;
         }
-        if (!prevProps.account.requiresTwoFactorAuth && this.props.account.requiresTwoFactorAuth) {
-            this.input2FA.focus();
+        setValidateCode(props.credentials.validateCode);
+    }, [props.credentials.validateCode, prevValidateCode]);
+
+    useEffect(() => {
+        if (!input2FARef.current || prevRequiresTwoFactorAuth || !props.account.requiresTwoFactorAuth) {
+            return;
         }
-    }
+        input2FARef.current.focus();
```

Comment:
Due to a regression, we'd need to change this to a `.clear()` instead to autofocus the 2FA input

Reply:
How is this a regression? The current code does focus the input. Can you please explain more on this

Reply:
I thought this was a new issue that cropped up since I wasn't able to find it, but looks like it's already reported: https://github.com/Expensify/App/issues/19711 -- the proposal seems to be to change to `clear` as well

Reply:
Let's not worry about it. It will be handled separately.